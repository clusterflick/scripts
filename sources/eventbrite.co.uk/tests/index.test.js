/** @jest-environment setup-polly-jest/jest-environment-node */
const { setupPolly, setupCacheMock } = require("../../../common/test-utils");
const { attributes, findEvents } = require("..");

const isRecording = false;

jest.mock("../../../common/cache");
setupCacheMock(__dirname, "2025-01-24");

const cinema = {
  name: "Genesis Cinema",
  geo: { lat: 51.52128726645794, lon: -0.051143457671891594 },
};

describe(attributes.name, () => {
  setupPolly(isRecording, __dirname);
  jest.useFakeTimers().setSystemTime(new Date("2025-01-24"));

  it(
    "find-events",
    async () => {
      const output = await findEvents(cinema);
      const data = JSON.parse(JSON.stringify(output));

      // Make sure the data looks roughly correct
      expect(data).toHaveLength(8);
      expect(data).toMatchSnapshot();
    },
    isRecording ? 120_000 : undefined,
  );
});
