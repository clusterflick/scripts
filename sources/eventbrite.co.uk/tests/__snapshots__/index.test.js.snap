// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Eventbrite find-events 1`] = `
[
  {
    "overview": {
      "actors": [],
      "categories": [],
      "directors": [],
      "duration": 9000000,
    },
    "performances": [
      {
        "accessibility": {},
        "bookingUrl": "https://www.eventbrite.com/checkout-external?eid=1140539557519",
        "notes": "",
        "status": {},
        "time": 1737828000000,
      },
    ],
    "title": "MYTH OF MAN: Special Early Screening - London 1/25",
    "url": "https://www.eventbrite.com/e/myth-of-man-special-early-screening-london-125-tickets-1140539557519",
  },
  {
    "overview": {
      "actors": [],
      "categories": [],
      "directors": [],
      "duration": 7200000,
    },
    "performances": [
      {
        "accessibility": {},
        "bookingUrl": "https://www.eventbrite.com/checkout-external?eid=1143433342909",
        "notes": "",
        "status": {},
        "time": 1739624400000,
      },
    ],
    "title": "Film Screening & Live Q&A: Every Good Marriage Begins with Tears",
    "url": "https://www.eventbrite.co.uk/e/film-screening-live-qa-every-good-marriage-begins-with-tears-tickets-1143433342909",
  },
  {
    "overview": {
      "actors": [],
      "categories": [],
      "directors": [],
      "duration": 12600000,
    },
    "performances": [
      {
        "accessibility": {},
        "bookingUrl": "https://www.eventbrite.com/checkout-external?eid=1144678878339",
        "notes": "",
        "status": {},
        "time": 1738953000000,
      },
    ],
    "title": "Life of SPOT 2 - The Journey Continues",
    "url": "https://www.eventbrite.co.uk/e/life-of-spot-2-the-journey-continues-tickets-1144678878339",
  },
  {
    "overview": {
      "actors": [],
      "categories": [],
      "directors": [],
      "duration": 10800000,
    },
    "performances": [
      {
        "accessibility": {},
        "bookingUrl": "https://www.eventbrite.com/checkout-external?eid=1050075844117",
        "notes": "",
        "status": {},
        "time": 1740850200000,
      },
    ],
    "title": "When Boys Don't Talk About Sex",
    "url": "https://www.eventbrite.co.uk/e/when-boys-dont-talk-about-sex-tickets-1050075844117",
  },
  {
    "overview": {
      "actors": [],
      "categories": [],
      "directors": [],
      "duration": 7200000,
    },
    "performances": [
      {
        "accessibility": {},
        "bookingUrl": "https://www.eventbrite.com/checkout-external?eid=1119625593289",
        "notes": "",
        "status": {},
        "time": 1740326400000,
      },
    ],
    "title": "London: Film premiere of The Wheels of Heaven / Дяволски игри 23.02.2025",
    "url": "https://www.eventbrite.co.uk/e/london-film-premiere-of-the-wheels-of-heaven-23022025-tickets-1119625593289",
  },
  {
    "overview": {
      "actors": [],
      "categories": [],
      "directors": [],
      "duration": 7200000,
    },
    "performances": [
      {
        "accessibility": {},
        "bookingUrl": "https://www.eventbrite.com/checkout-external?eid=1145691426899",
        "notes": "",
        "status": {},
        "time": 1739023200000,
      },
    ],
    "title": "London: Film premiere of "Vasko and The Violin" - 08.02.25",
    "url": "https://www.eventbrite.co.uk/e/london-film-premiere-of-vasko-and-the-violin-080225-tickets-1145691426899",
  },
  {
    "overview": {
      "actors": [],
      "categories": [],
      "directors": [],
      "duration": 12600000,
    },
    "performances": [
      {
        "accessibility": {},
        "bookingUrl": "https://www.eventbrite.com/checkout-external?eid=1215578761799",
        "notes": "",
        "status": {},
        "time": 1740157200000,
      },
    ],
    "title": "Premiere of 'Behind The Scene' Documentary and Workshops",
    "url": "https://www.eventbrite.co.uk/e/premiere-of-behind-the-scene-documentary-and-workshops-tickets-1215578761799",
  },
  {
    "overview": {
      "actors": [],
      "categories": [],
      "directors": [],
      "duration": 6300000,
    },
    "performances": [
      {
        "accessibility": {},
        "bookingUrl": "https://www.eventbrite.com/checkout-external?eid=1147883443279",
        "notes": "",
        "status": {},
        "time": 1738001700000,
      },
    ],
    "title": "Hercules Nicolaou- 'The Rebirth'",
    "url": "https://www.eventbrite.co.uk/e/hercules-nicolaou-the-rebirth-tickets-1147883443279",
  },
]
`;
